src/ng/templateRequest.js

```diff
@@ -45,12 +79,10 @@ function $TemplateRequestProvider() {
         transformResponse = null;
       }
 
-      var httpOptions = {
-        cache: $templateCache,
-        transformResponse: transformResponse
-      };
-
-      return $http.get(tpl, httpOptions)
+      return $http.get(tpl, extend({
+          cache: $templateCache,
+          transformResponse: transformResponse
+        }, httpOptions))
```

Comment:
The point of modifying the `transformResponse` is that the default `$http` service adds in a JSON transformation, which we don't want in `$templateRequest`.

If we are going to allow people to configure the options then we should allow them to override everything, including this `transformResponse` and also to provide their own cache if we really wanted them to; although the same could be achieved by overriding the `$templateCache` service itself.

I can add a test for overriding the `cache` but it is really just the same code path as overriding the `transformResponse`.